code/clue.js

```diff
@@ -1,59 +1,220 @@
+const changeButton = () => {
+  mysteryButtonElement.innerHTML = "?"
+
+  if (mystery.killer !== null && mystery.weapon !== null && mystery.room !== null) {
+    mysteryButtonElement.style.background = "#EFF066"
+  } else {
+    mysteryButtonElement.style.background = "transparent"
+  }
+}
+
+//*** FUNCTION FOR KILLER CARD ***//
 
-// This function will be invoked when you click on the killer card.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  const killerAgeElement = document.getElementById("killerAge")
+  const killerOccupationElement = document.getElementById("killerOccupation")
+
+  mystery.killer = randomSelector(killers)
+
+  killerNameElement.innerHTML = `${mystery.killer.fullName()}`
+  killerAgeElement.innerHTML = `Age: ${mystery.killer.age}`
+  killerOccupationElement.innerHTML = `Occupation: ${mystery.killer.occupation}`
+
+  changeCard("killerCard")
+  changeButton()
+}
+
+//*** FUNCTION FOR WEAPON CARD ***//
+
+const pickWeapon = () => {
+
+  const weaponWeightElement = document.getElementById("weaponWeight")
+
+  mystery.weapon = randomSelector(weapons)
+
+  weaponNameElement.innerHTML = `${mystery.weapon.name}`
+  weaponWeightElement.innerHTML = `Weight: ${mystery.weapon.weight} kg`
+
+  changeCard("weaponCard")
+  changeButton()
+}
+
+//*** FUNCTION FOR ROOM CARD ***//
+
+const pickRoom = () => {
+
+  const roomColorElement = document.getElementById("roomColor")
+
+  mystery.room = randomSelector(rooms)
+
+  roomNameElement.innerHTML = `${mystery.room.name}`
+  roomColorElement.innerHTML = `Color: ${mystery.room.color}`
+
+  changeCard("roomCard")
+  changeButton()
+}
+
+//*** FUNCTION THAT REVEALS THE MYSTERY ***//
+//invoked by button onclick in index.html
+
+const revealMystery = () => {
+
+  const cardElements = document.querySelectorAll(".card")
+  const nameAndInfoElements = [
+    ...document.querySelectorAll(".name"),
+    ...document.querySelectorAll(".info")
+  ]
+
+
+  const revealMysteryNow = () => {
+    mysteryButtonElement.innerHTML = (`It was ${mystery.killer.fullName()} with a ${mystery.weapon.name} in the ${mystery.room.name}.`)
+    mysteryButtonElement.style.background = "transparent"
+    mysteryButtonElement.style.opacity = 1.0;
+    nameAndInfoElements.forEach(info => info.innerHTML = "")
+    cardElements.forEach(card => card.style.background = "#30F2C6")
+    Object.keys(mystery).forEach(key => mystery[key] = null)
+  }
+
+  const pickCard = () => {
+    if (mystery.killer === null) {
```

Comment:
Another tip for you Annika, we can simplify this statement, let me introduce you to one concept :
`null` values in conditionals are somehow "converted" by JavaScript to boolean values. What does that mean? Well, for example that `null` value is converted by JS into `false` value. So :
```
if (mystery.killer)
```
For JavaScript will look like this :
```
if (false)
```
There is also addition to that - there is a special sign in JavaScript `!`. What does it do? It changes `boolean` value to the opposite one. So, combining our knowledge we can assume :
```
null is converted by JS to false
```
```
!null is converted by JS to true
```
How can we use that? Clean code conventions says that rather than writing
```
if (mystery.killer === null)
```
we should write
```
if (!mystery.killer)
```

There are actually more of this kind of values in JS. For example empty string `""`, zero number `0` or `undefined` also are converted to `false`, and using `!` before them will convert all expression to `true`.
Non-empty strings however, as well as any numbers different than `0`, objects and arrays are converted to true.
For further reference about that please check out http://speakingjs.com/es5/ch10.html

Reply:
Love it, thanks for a very extensive and clear answer, obviously I want to use this instead!